### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
## PR Summary

Simple documentation pr, explains that forked branches will not automatically trigger the CI. I also added some internal
file links to ease navigation of the Contributing.md file. 

## PR Checklist

- [x] Changes are summarized in CHANGELOG.md

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -73,7 +91,17 @@ In order for code to be merged into `develop` it must
 The reviewers are expected to look out for the items above before approving a merge
 request.
 
-#### Linting Code
+#### Merging Code from a Fork
+
+Unfortunately, the CI will not automatically trigger for forks. This is for security
```
Comment (pgrete) [inline CONTRIBUTING.md:97]:
```suggestion
PRs can opened as usual from forks.
Unfortunately, the CI will not automatically trigger for forks. This is for security
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -73,7 +91,17 @@ In order for code to be merged into `develop` it must
...
+reasons. As a workaround, in order to trigger the CI, a local branch will need to be created
+on Parthenon first. The forked code can then be merged into the local branch on 
+Parthenon. At this point when a new merge request is opened from the local branch
+to the develop branch it will trigger the CI. 
+
```
Comment (pgrete) [inline CONTRIBUTING.md:112]:
```suggestion
Someone of the Parthenon core team will take care of the work around once a PR from a fork.
No extra work is required from the contributor.

```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -73,7 +91,17 @@ In order for code to be merged into `develop` it must
...
+NOTE: Any subsequent updates made to the forked branch will need to be manually pulled into the local branch.
```
Comment (pgrete) [inline CONTRIBUTING.md:113]:
```suggestion
The workaround workflow for the Parthenon core developer may look like
(from a local Parthenon repository pulling in changes from a `feature-A` branch in a fork):

$ git remote add external-A https://github.com/CONTRIBUTOR/parthenon.git
$ git fetch external-A
$ git checkout external-A/feature-A
$ git push --set-upstream origin CONTRIBUTOR/feature-A

NOTE: Any subsequent updates made to the forked branch will need to be manually pulled into the local branch. 
```

**Review by pgrete — APPROVED** [review]
I slightly extended the description with an example. Otherwise, looks good to go. Thanks!

**Review by Yurlungur — APPROVED** [review]
I approve with @pgrete 's modifications. Thanks for doing this!

**Comment by JoshuaSBrown** [comment]
@jlippuner ping

**Pull request merged**